### PR TITLE
Mcarrera commodity forwards update expectations PROD

### PR DESCRIPTION
**Pull request by PayalKhanna**
original contribution #2709

### CODE REVIEW

**Comment by PayalKhanna** [comment]
Issue: https://github.com/finos/common-domain-model/issues/2695

**Review by lolabeis — COMMENTED** [review]
@manucarreramoreno Do we have samples in the test pack that illustrate the changes, or could we add some?

**Comment by manucarreramoreno** [comment]
@lolabeis we have some FpML samples but I have not been able to figure out the mappings

**Comment by manucarreramoreno** [comment]
@lolabeis sure, we are already tracking it in AT

**Review by eacunaISDA — APPROVED** [review]

**Comment by PayalKhanna** [comment]
@lolabeis are you happy with the changes to be merged

File: rosetta-source/src/main/rosetta/product-qualification-func.rosetta [hunk 1]
```diff
@@ -1531,3 +1536,18 @@ func Qualify_Commodity_Swaption: <"Qualifies a product as a Swaption that can be
...
+    output:
+        is_product boolean (1..1)
+            [synonym ISDA_Taxonomy_v1 value "Commodity_Spot_Fwd"]
+            [synonym ISDA_Taxonomy_v2 value "Commodity_Forward"]
+    set is_product:
+        Qualify_AssetClass_Commodity(economicTerms) = True
+            and (
+                (economicTerms -> payout -> forwardPayout, economicTerms -> payout -> fixedPricePayout) only exists or // Fixed Price Forward
+                (economicTerms -> payout -> forwardPayout, economicTerms -> payout -> commodityPayout) only exists) // Floating Price Forward
+            and economicTerms -> payout -> forwardPayout -> underlier -> commodity exists
```
[the remark this answers is not in the document]
Comment (manucarreramoreno) [reply rosetta-source/src/main/rosetta/product-qualification-func.rosetta:1553]: This line is consistent with the last line in `Qualify_Commodity_Option` that also checks that the option underlier is a commodity even though the `Qualify_AssetClass_Commodity` already does it

[the remark this answers is not in the document]
Comment (lolabeis) [reply rosetta-source/src/main/rosetta/product-qualification-func.rosetta:1553]: In this case that line is redundant as well in `Qualify_Commodity_Option` - and so is it in `Qualify_Commodity_Swap_FixedFloat`.

To be on the safe side for DRR, let's do as follows:
- Remove the redundant line from `Qualify_Commodity_Forward` only
- Leave `Qualify_Commodity_Option` and `Qualify_Commodity_Swap_FixedFloat` as they are, for now
- In a subsequent development, remove the redundant line from `Qualify_Commodity_Option` and `Qualify_Commodity_Swap_FixedFloat` and check that indeed no impact on DRR

Please make the change in both the dev and prod PRs.

[the remark this answers is not in the document]
Comment (manucarreramoreno) [reply rosetta-source/src/main/rosetta/product-qualification-func.rosetta:1553]: Done, for both PRs, so they should be good to go

**Review by lolabeis — APPROVED** [review]

**Comment by Oblongs** [comment]
Reviewed and approved in the [CDM Derivatives Products and Business Events Working Group - February 27th, 2024](https://github.com/finos/common-domain-model/issues/2669#top)

**Pull request merged**